client2/src/components/game/gui/chat_input/ChatInput.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect, useState } from "react";
+import "./ChatInput.scss";
+import PubSub from "pubsub-js";
+import { CHAT_CLOSE, CHAT_OPEN, ENTER_KEY } from "../../../../shared/EventTypes";
+
+function ChatInput() {
+    const [chatStatus, setChatStatus] = useState(CHAT_CLOSE);
+    const [chatMsg, setChatMsg] = useState("");
+
+    const clearChatMsg = () => {
+        setChatMsg("");
+    };
+
+    const sendChatMsg = (webSocket, msg) => {
+        webSocket.sendEvent("chat", msg);
+    };
+
+    const hideChatInput = () => {
+        PubSub.publish(CHAT_CLOSE, null);
+        setChatStatus(CHAT_CLOSE);
+    };
+
+    const showChatInput = () => {
+        PubSub.publish(CHAT_OPEN, null);
+        setChatStatus(CHAT_OPEN);
+    };
+
+    useEffect(() => {
+        const subs = [
+            // hitPoints is undefined for now, until player is implemented again
+            PubSub.subscribe(ENTER_KEY, (msg, { hitPoints }) => {

```

Comment:
hitPoints can reference `PlayerState.hitPoints`

Reply:
Oh I totally missed that, working on a patch now.

Reply:
Refactored at a2eb9e5